core/client/fs/src/main/java/alluxio/client/file/FileSystemContext.java

```diff
@@ -369,10 +391,18 @@ public BlockWorkerClient acquireBlockWorkerClient(final WorkerNetAddress workerN
   public void releaseBlockWorkerClient(WorkerNetAddress workerNetAddress,
       BlockWorkerClient client) {
     SocketAddress address = NetworkAddressUtils.getDataPortSocketAddress(workerNetAddress);
-    try {
-      client.close();
-    } catch (IOException e) {
-      LOG.warn("Error closing block worker client for address {}", address, e);
+    ClientPoolKey key = new ClientPoolKey(address,
+        SaslParticipantProviderUtils.getImpersonationUser(mParentSubject));
+    if (mBlockWorkerClientPool.containsKey(key)) {
+      mBlockWorkerClientPool.get(key).release(client);
```

Comment:
These two commands will not be atomic, perhaps use `computeIfPresent`? Or leave a TODO since I don't think that is necessarily clean either.

Reply:
The key will not be removed from the map so no synchronization is necessary.